journal.py

```diff
@@ -0,0 +1,233 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import os
+from pyramid.config import Configurator
+from pyramid.view import view_config
+from waitress import serve
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+import datetime
+"""
+sessionmaker is a factory that makes factories
+  Session = sessionmaker(bind=engine)
+    recall that Session is a class being defined as a sessionmaker
+    class but where the bind is equal to the engine variable that
+    should be defined elsewhere; the engine is the connection, the
+    Session is the cursor.
+  session = Session()
+    actually starts a session
+scoped_session is similar to that process above, only a scoped
+session helps to maintain the association of one session with one
+request, even when there are many many requests happening all at once.
+"""
+from sqlalchemy.orm import sessionmaker, scoped_session
+from zope.sqlalchemy import ZopeTransactionExtension
+from pyramid.httpexceptions import HTTPFound
+from sqlalchemy.exc import DBAPIError
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+from cryptacular.bcrypt import BCRYPTPasswordManager
+from pyramid.security import remember, forget
+
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+
+"""
+bind a symbol, available to all the code in the project, at the
+module scope which will be responsible creating session for each
+request. it is the point of access to the database.
+"""
+
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+
+
+Base = declarative_base()
+
+
+class Entry(Base):
+    __tablename__ = "entries"
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.created.desc()).all()
+
+    @classmethod
+    def write(cls, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls(title=title, text=text)
+        session.add(instance)
+        return instance
+
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(255), nullable=False)
+    created = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow
+    )
+    text = sa.Column(sa.UnicodeText, nullable=False)
```

Comment:
It is customary to put "fields" on an ORM model at the top of the class definition, followed by class methods and instance methods.  Placing your field definitions here will defeat expectations and confuse others.